controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -334,14 +334,19 @@ public OperationContext createContext(String scope, String name) {
                 new AbstractMap.SimpleEntry<>(x.getKeyStart(), x.getKeyEnd())).collect(Collectors.toList());
 
         future.thenAccept(result -> {
-            DYNAMIC_LOGGER.incCounterValue(nameFromStream(SEGMENTS_COUNT, scope, name),
+            DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(SEGMENTS_COUNT, scope, name),
                     newSegments.size() - sealedSegments.size());
             getSealedRanges(scope, name, sealedSegments, context, executor)
                     .thenAccept(sealedRanges -> {
-                        DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(SEGMENTS_SPLITS, scope, name),
-                                findSplits(sealedRanges, newRanges));
-                        DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(SEGMENTS_MERGES, scope, name),
-                                findSplits(newRanges, sealedRanges));
+                        long oldNumSplits = DYNAMIC_LOGGER.getCounterValue(nameFromStream(SEGMENTS_SPLITS, scope, name));
+                        long newNumSplits = findSplits(sealedRanges, newRanges);
+                        DYNAMIC_LOGGER.incCounterValue(nameFromStream(SEGMENTS_SPLITS, scope, name),
```

Comment:
@shiveshr The method incCounterValue takes input a delta, hence adding the delta value to counter. I am just making sure we are not adding negative values to splits, since it is accumulated number of splits. If oldNumSplits > newNumSplits just adding the newNumSplits value.